101-vsts-cloudloadtest-rig/azuredeploy.json

```diff
@@ -0,0 +1,226 @@
+               "adminUsername":"[parameters('adminUsername')]",
+               "adminPassword":"[parameters('adminPassword')]"
+            },
+            "storageProfile":{
+               "imageReference":{
+                  "publisher":"[variables('imagePublisher')]",
+                  "offer":"[variables('imageOffer')]",
+                  "sku":"[variables('windowsOSVersion')]",
+                  "version":"latest"
+               },
+               "osDisk":{
+                  "name":"osdisk",
+                  "vhd":{
+                     "uri":"[concat('http://',variables('storageAccountName'),'.blob.core.windows.net/',variables('vmStorageAccountContainerName'),'/',concat(variables('vmName'), copyIndex(), variables('OSDiskName')),'.vhd')]"
+                  },
+                  "caching":"ReadWrite",
+                  "createOption":"FromImage"
+               },
+               "dataDisks":[
+                  {
+                     "name":"datadisk",
+                     "diskSizeGB":"[variables('sizeOfDiskInGB')]",
+                     "lun":0,
+                     "vhd":{
+                        "uri":"[concat('http://',variables('storageAccountName'),'.blob.core.windows.net/',variables('vmStorageAccountContainerName'),'/', concat(variables('vmName'), copyIndex(), variables('dataDisk1VhdName')),'.vhd')]"
+                     },
+                     "createOption":"Empty"
+                  }
+               ]
+            },
+            "networkProfile":{
+               "networkInterfaces":[
+                  {
+                     "id":"[resourceId('Microsoft.Network/networkInterfaces',concat(variables('nicName'), 'i', copyIndex()))]"
+                  }
+               ]
+            },
+            "diagnosticsProfile":{
+               "bootDiagnostics":{
+                  "enabled":"true",
+                  "storageUri":"[concat('http://',variables('storageAccountName'),'.blob.core.windows.net')]"
+               }
+            }
+         }
+      },
+      {
+         "type":"Microsoft.Compute/virtualMachines/extensions",
+         "copy":{
+            "name":"vmCopy",
+            "count":"[parameters('agentCount')]"
+         },
+         "name":"[concat(variables('vmName'), 'i', copyIndex(), '/StartupScript')]",
+         "apiVersion":"2015-06-15",
+         "location":"[resourceGroup().location]",
+         "dependsOn":[
+            "[concat('Microsoft.Compute/virtualMachines/',concat(variables('vmName'), 'i', copyIndex()))]"
+         ],
+         "properties":{
+            "publisher":"Microsoft.Compute",
+            "type":"CustomScriptExtension",
+            "forceUpdateTag":"[variables('sequenceVersion')]",
+            "typeHandlerVersion":"1.8",
+            "autoUpgradeMinorVersion":true,
+            "settings":{
+               "fileUris":[
+                  "https://elsprodch2su1.blob.core.windows.net/ets-containerfor-loadagentresources/bootstrap/ConfigureUserLoadAgent.ps1"
```

Comment:
Kay, this script is being pulled from our production deployment storage. This script is expected to change across our production deployments. We have designed this template to be very thin and to be static (most of the time) to avoid the need of frequent deployments. 
Given this, I'll prefer not to push the script in GitHub.